From af871abd542a60b8e71a3030fa264432daabd4df Mon Sep 17 00:00:00 2001
From: Marcin Zyla <marcin.zyla@vprocess.pl>
Date: Tue, 16 Sep 2025 12:59:31 +0200
Subject: [PATCH] fix: deleting interfaces too fast causes stat segment
 assertion

fixes:
/vpp-src/src/vlib/stats/collector.c:88 (update_node_counters) assertion `node_data[i].symlinks[j] != CLIB_U32_MAX' fails"
---
 src/vnet/interface.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/vnet/interface.c b/src/vnet/interface.c
index db47d42c8..68c1f121d 100644
--- a/src/vnet/interface.c
+++ b/src/vnet/interface.c
@@ -1149,9 +1149,9 @@ vnet_delete_hw_interface (vnet_main_t * vnm, u32 hw_if_index)
 	}
 
       vlib_node_rename (vm, hw->output_node_index,
-			"interface-%d-output-deleted", hw_if_index);
-      vlib_node_rename (vm, hw->tx_node_index, "interface-%d-tx-deleted",
-			hw_if_index);
+			"interface-%d-%d-output-deleted", hw->output_node_index, hw_if_index);
+      vlib_node_rename (vm, hw->tx_node_index, "interface-%d-%d-tx-deleted", 
+			hw->tx_node_index, hw_if_index);
       vlib_unregister_errors (vm, hw->output_node_index);
       vlib_unregister_errors (vm, hw->tx_node_index);
       vec_add2 (im->deleted_hw_interface_nodes, dn, 1);
-- 
2.51.0

